### DISCUSSION TITLE
Anyone understand how to use "ghost attention" (GAtt) with llama 2?

### DISCUSSION THREAD

**Discussion by JohanAR** [ggml-org/llama.cpp#2541] [General]
Chapter 3.3 of "Llama 2: Open Foundation and Fine-Tuned Chat Models" talks about using GAtt to avoid having to repeat instructions in multi-turn dialogue. Is this something that would need to be implemented in llama.cpp, or does it already exist?

**Comment by WeirdConstructor** [comment]
As far as I understood: Gatt is a special algorithm that was used when LLaMA 2 Chat was fine tuned. The GAtt is not something that requires a change in how inference is done.

**Reply by JohanAR** [reply]
Ah, ok. I was thinking that there might be something one needed to take into consideration when writing prompts, or choosing which tokens to send to the model.

**Reply by darkman111a** [reply]
actually to utilize GAtt you must insert <<SYS>> & <</SYS>> between any text  to modify the behavior of the entire session. Your intution was correct.
